taskvine/src/manager/vine_task.c

```diff
@@ -435,23 +435,23 @@ void vine_task_add_output( struct vine_task *t, struct vine_file *f, const char
 	list_push_tail(t->output_mounts, m);
 }
 
-void vine_task_add_input_file(struct vine_task *t, const char *local_name, const char *remote_name, vine_file_flags_t flags)
+void vine_task_add_input_file(struct vine_task *t, const char *local_name, const char *remote_name, vine_mount_flags_t flags)
 {
-	struct vine_file *f = vine_file_local(local_name);
+	struct vine_file *f = vine_file_local(local_name, (vine_file_flags_t)flags);
```

Comment:
Careful!  A cast is usually a sign of a problem.

`vine_file_flags_t` and `vine_mount_flags_t` are not the same thing.

Here, we are creating a `vine_file_local` which takes file flags, and attaching it to a task with `vine_task_add_input` which takes mount flags.

The `vine_file_local` must be created with **file** flags, which haven't been given here.

Since this is a shortcut method that doesn't allow one to specify the same file used multiple times, I think the right thing to do here is pass in file flags indicating no-share and no-cache.

(Similar comments apply to the other `add_input_X` methods.)

Reply:
We intend to remove these "add_input_X" methods, so I cast the flags to get rid of the warnings for the time being, since the VINE_CACHE symbols are still in both enumerations it still works.

Reply:
That's a grenade waiting to go off.  :). Please put in some explicit default values there.